presto-main/src/main/java/com/facebook/presto/operator/aggregation/MaxBy.java

```diff
@@ -123,6 +123,10 @@ public static void combine(MaxOrMinByState state, MaxOrMinByState otherState)
             state.setKey(otherState.getKey());
             state.setValue(otherState.getValue());
         }
+        else if (otherState.getKey() == null) {
+            otherState.setKey(state.getKey());
```

Comment:
this is unnecessary. otherState doesn't need to be updated, so you can just return here